src/toil/leader.py

```diff
@@ -335,6 +336,10 @@ def mainLoop(config, batchSystem, jobStore, rootJobWrapper):
     logger.info("Found %s jobs to start and %i jobs with successors to run",
                 len(toilState.updatedJobs), len(toilState.successorCounts))
 
+    #Set proper environment variables
+    for v in getEnv():
```

Comment:
getEnv() should be a method of the batch system class. It can't be a statically defined function in the module. Look at the other methods in AbstractBatchSystem and model your new method after those.